packages/gluestick/shared/containers/Root.js

```diff
@@ -73,28 +73,39 @@ export default class Root extends Component<DefaultProps, Props, State> {
   }
 
   _renderRouter(props: Props): Object | Component<*, *, *> {
-    const {
-      routes,
-      routerContext,
-      routerHistory,
-    } = props;
-
     // server rendering
-    if (routerContext) return routerContext;
+    if (props.routerContext) return props.routerContext;
 
     // router middleware
     const render: Function = applyRouterMiddleware(
-      useScroll((prevRouterProps, results) => {
+      useScroll((prevRouterProps, { location, routes }) => {
+        // If the user provides custom scroll behaviour, use it, otherwise fallback to the default
+        // behaviour.
+        const { useScroll: customScrollBehavior } = routes.find(route => (
+          route.useScroll &&
+          prevRouterProps &&
+          prevRouterProps.location.pathname !== location.pathname
+        )) || {};
+
+        if (typeof customScrollBehavior === 'function') {
+          return customScrollBehavior(prevRouterProps, location);
+        }
```

Comment:
Have you seen how helpless are stack traces for some errors? It's better to have that check and add `else` block with meaningful error message - so just by reading it, you would know what you have to fix.